utils/zerovec/benches/vzv.rs

```diff
@@ -0,0 +1,153 @@
+    // Lcg64Xsh32 is a small, fast PRNG for reproducible benchmarks.
+    let mut rng1 = Lcg64Xsh32::seed_from_u64(seed);
+    let mut rng2 = Lcg64Xsh32::seed_from_u64(rand::Rng::gen(&mut rng1));
+    let alpha_dist = Alphanumeric;
+    let len_dist = Uniform::from(lengths);
+    let string_vec = len_dist
+        .sample_iter(&mut rng1)
+        .take(count)
+        .map(|len| {
+            (&alpha_dist)
+                .sample_iter(&mut rng2)
+                .take(len)
+                .map(char::from)
+                .collect::<String>()
+        })
+        .collect();
+    (string_vec, rand::Rng::gen(&mut rng1))
+}
+
+fn overview_bench(c: &mut Criterion) {
+    // Same as vzv/char_count/vzv but with different inputs
+    let seed = 42;
+    let (string_vec, _) = random_alphanums(2..=10, 100, seed);
+    let bytes: Vec<u8> = VarZeroVec::get_serializable_bytes(string_vec.as_slice()).unwrap();
+    let vzv = VarZeroVec::<String>::try_from_bytes(black_box(bytes.as_slice())).unwrap();
+
+    c.bench_function("vzv/overview", |b| {
+        b.iter(|| {
+            black_box(&vzv)
+                .iter()
+                .fold(0, |sum, string| sum + string.chars().count())
+        });
+    });
+
+    #[cfg(feature = "bench")]
+    {
+        char_count_benches(c);
+        binary_search_benches(c);
+    }
+
+    #[cfg(all(feature = "bench", feature = "serde"))]
+    {
+        serde_benches(c);
+    }
+}
+
+#[cfg(feature = "bench")]
+fn char_count_benches(c: &mut Criterion) {
+    let seed = 2021;
+    let (string_vec, _) = random_alphanums(2..=20, 100, seed);
+    let bytes: Vec<u8> = VarZeroVec::get_serializable_bytes(string_vec.as_slice()).unwrap();
+    let vzv = VarZeroVec::<String>::try_from_bytes(black_box(bytes.as_slice())).unwrap();
+
+    // *** Count chars in vec of 100 strings ***
+    c.bench_function("vzv/char_count/slice", |b| {
+        b.iter(|| {
+            black_box(&string_vec)
+                .iter()
+                .fold(0, |sum, string| sum + string.chars().count())
+        });
+    });
+
+    // *** Count chars in vec of 100 strings ***
+    c.bench_function("vzv/char_count/vzv", |b| {
+        b.iter(|| {
+            black_box(&vzv)
+                .iter()
+                .fold(0, |sum, string| sum + string.chars().count())
+        });
+    });
+}
+
+#[cfg(feature = "bench")]
+fn binary_search_benches(c: &mut Criterion) {
+    let seed = 2021;
+    let (string_vec, seed) = random_alphanums(2..=20, 500, seed);
+    let (needles, _) = random_alphanums(2..=20, 10, seed);
+    let bytes: Vec<u8> = VarZeroVec::get_serializable_bytes(string_vec.as_slice()).unwrap();
+    let vzv = VarZeroVec::<String>::try_from_bytes(black_box(bytes.as_slice())).unwrap();
+
+    // *** Binary search vec of 500 strings 10 times ***
+    c.bench_function("vzv/binary_search/slice", |b| {
+        b.iter(|| {
+            black_box(&needles)
```

Comment:
issue: All of the inputs need to be `black_box`ed. Alternatively, use `bench_with_input`. We should probably do this in the other benchmarks too, I was doing it locally but I don't recall if it got upstreamed

Reply:
I added more black boxes; did I miss any?

I don't like bench_with_input because it creates a new bench for every input, which is slow and unnecessary.

Reply:
Looks good! And yeah that's fine; perhaps we should ask criterion to create a version of `bench_function` that behaves like with_input